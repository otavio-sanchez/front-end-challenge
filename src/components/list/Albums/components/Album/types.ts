import { Album } from '../../types';

export interface Props extends Album {
    height?: number;
}
