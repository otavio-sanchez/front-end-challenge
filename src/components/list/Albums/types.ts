export interface Props {
    data: Album[];
    title: string;
    height?: number;
}

export interface Album {
    image: string;
    redirect: string;
    id: string;
    name: string;
    artist: string;
}
