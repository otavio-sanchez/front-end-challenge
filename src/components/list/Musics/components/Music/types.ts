export interface Props {
    id: string;
    name: string;
    time: number;
    trackNumber: number;
    onClick: Function;
}
