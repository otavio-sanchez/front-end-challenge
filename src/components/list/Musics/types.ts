export interface Props {
    data: Music[];
    onClickItem: Function;
}

export interface Music {
    id: string;
    name: string;
    time?: number;
    preview_url: string;
    track_number: number;
    duration_ms?: number;
}
