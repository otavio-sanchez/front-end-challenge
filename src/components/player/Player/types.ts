export interface Props {
    images: Image[];
    id: string;
    name: string;
    artists: Artist[];
    onClick: Function;
    preview: string;
}

export interface Artist {
    id: string;
    name: string;
}

export interface Image {
    height: number;
    url: string;
    width: number;
}
