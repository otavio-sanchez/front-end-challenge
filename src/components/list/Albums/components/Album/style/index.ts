import styled, { keyframes } from 'styled-components';
import sizes from '../../../../../../assets/theme/sizes';

const opacityAnimation = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
`;

const Content = styled.article`
    width: 100%;
    cursor: pointer;
    animation: ${opacityAnimation} 1s linear;
`;

const Image = styled.div<{ src: string; height?: number }>`
    
    width: 100%;
    height: ${({ height }): number => (height ? height : 250)}px;
    
    @media (max-width: ${sizes.screen.tabletWidth + 1}px) {
        height: ${({ height }): number => (height ? height : 500)}px;
    }
    background-image: url('${({ src }): string => src}');
    background-size: cover;
    background-position: center;
    transition: opacity ease-in-out 0.5s;
    opacity: 1;
    &:hover {
        transition: opacity ease-in-out 0.5s;
        opacity: 0.5;
    }
`;

const Infos = styled.div`
    text-align: center;
    padding: 20px 5%;
    width: 100%;
    box-sizing: border-box;
`;

export { Content, Image, Infos };
