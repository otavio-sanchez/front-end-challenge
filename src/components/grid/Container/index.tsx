import * as React from 'react';
import { ContainerContent } from './style';
import { Props } from './types';

const Container = ({ children, maxWidth }: Props): JSX.Element => {
    return <ContainerContent maxWidth={maxWidth}>{children}</ContainerContent>;
};

export default Container;
