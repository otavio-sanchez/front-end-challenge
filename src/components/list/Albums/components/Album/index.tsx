import * as React from 'react';
import { Text } from '../../../../typography';
import { Content, Image, Infos } from './style';
import { Props } from './types';

const Album = ({ image, name, height, artist, onClick }: Props): JSX.Element => (
    <Content onClick={(): Function => onClick()}>
        <Image src={image} height={height} />
        <Infos>
            <Text type="subtitle">
                <>{name}</>
            </Text>
            <Text type="regular">
                <>{artist}</>
            </Text>
        </Infos>
    </Content>
);

export default Album;
