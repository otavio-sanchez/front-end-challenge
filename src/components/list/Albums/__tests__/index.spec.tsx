import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Albums } from '../..';
import { Props } from '../types';
import albums from '../../../../constants/mocks/albums';
import { Container } from '../../../grid';
import Album from '../components/Album';
import { Text } from '../../../Typography';

describe('Albums', () => {
    const props: Props = {
        data: albums,
        title: 'MetÃ¡lica',
        height: 360,
        onClickItem: () => {},
        loading: false,
    };

    it('renders without crashing given the required props', () => {
        const wrapper = shallow(<Albums {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should not render component when was loading', () => {
        const wrapper = shallow(<Albums {...props} />);
        wrapper.setProps({ loading: true });
        expect(wrapper.find(Container)).toHaveLength(0);
    });

    it('should render three Album component', () => {
        const wrapper = shallow(<Albums {...props} />);
        expect(wrapper.find(Album)).toHaveLength(3);
    });

    it('should render feedback when there is not data', () => {
        const wrapper = shallow(<Albums {...props} />);
        wrapper.setProps({ data: [] });
        expect(wrapper.find(Text).get(1).props.children).toContain('Nada encontrado.');
    });
});
