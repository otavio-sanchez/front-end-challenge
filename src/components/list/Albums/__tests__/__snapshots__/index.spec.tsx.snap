// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums renders without crashing given the required props 1`] = `
<Fragment>
  <Container>
    <Row
      columnsDesktop={1}
      columnsMobile={1}
      columnsTablet={1}
      spacingColumn={30}
      spacingRow={30}
    >
      <Col>
        <Text
          type="title"
        >
           
          Met√°lica
           
        </Text>
      </Col>
    </Row>
    <Row
      columnsDesktop={5}
      columnsMobile={1}
      columnsTablet={2}
      spacingColumn={30}
      spacingRow={30}
    >
      <Col
        key="0"
      >
        <Album
          artists={
            Array [
              Object {
                "id": "0",
                "name": "artist foo",
              },
            ]
          }
          height={360}
          id="0"
          images={
            Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00004851af07dc851962508661bbcfce",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 64,
              },
            ]
          }
          name="Album foo"
          onClick={[Function]}
        />
      </Col>
      <Col
        key="1"
      >
        <Album
          artists={
            Array [
              Object {
                "id": "1",
                "name": "artist bar",
              },
            ]
          }
          height={360}
          id="1"
          images={
            Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00004851af07dc851962508661bbcfce",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 64,
              },
            ]
          }
          name="Album bar"
          onClick={[Function]}
        />
      </Col>
      <Col
        key="2"
      >
        <Album
          artists={
            Array [
              Object {
                "id": "2",
                "name": "artist baz",
              },
            ]
          }
          height={360}
          id="2"
          images={
            Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00004851af07dc851962508661bbcfce",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 64,
              },
            ]
          }
          name="Album baz"
          onClick={[Function]}
        />
      </Col>
    </Row>
  </Container>
</Fragment>
`;
